cmds.go

```diff
@@ -49,6 +72,75 @@ type SonyCmdToggle struct {
 	Input string `short:"i" help:"Specify host input, do not autodetect"`
 }
 
+// AfterApply creates a new [Screen] from the flags in the [screenFlags] struct.
+func (sf *screenFlags) AfterApply() error {
+	s, err := NewScreen(sf.Display, sf.Manufacturer, sf.ProductCode)
+	if err != nil {
+		return err
+	}
+	sf.screen = s
+	return nil
+}
+
+// Run (offscreen run) runs offscreen to turn the connected TV on and off
+// in line with X screen saver events.
+func (cmd *RunCmd) Run() error {
+	ch := make(chan bool)
+	g := errgroup.Group{}
+	g.Go(func() error { return cmd.screen.Watch(ch) })
+	g.Go(func() error { return cmd.loop(ch) })
+	return g.Wait()
+}
+
+func (cmd *RunCmd) loop(ch <-chan bool) error {
+	defer cmd.screen.Close()
+	c := NewRESTClient(cmd.Hostname, cmd.PSK)
+	ourLabel, err := hostnameLabel()
+	if err != nil {
+		return fmt.Errorf("could not get input URI for %s: %w", cmd.Input, err)
+	}
+	labels, err := c.Inputs()
+	if err != nil {
+		return err
+	}
+	ourInput, ok := labels[ourLabel]
+	if !ok {
+		return fmt.Errorf("tv set does not have labelled input: %s", ourLabel)
+	}
+	cmd.Input = ourInput
+
+	for ssOn := range ch {
+		status, err := c.PowerStatus()
+		if err != nil {
+			return fmt.Errorf("could not get power status: %w", err)
+		}
+		if status == "standby" && ssOn {
+			continue
+		}
+		if status == "standby" && !ssOn {
+			if err := c.SetPowerStatus(true); err != nil {
+				return fmt.Errorf("could not set power status: %w", err)
+			}
+		}
+		input, err := c.SelectedInput()
+		if err != nil {
+			return fmt.Errorf("could not get selected input: %w", err)
+		}
+		if status == "standby" && !ssOn && input != cmd.Input {
```

Comment:
i'm a bit lost here. I select a different Input (computer) via cmd.Input. why does it matter what the current screensaver status is?

Reply:
i think it would be helpful to have an overview before the loop starts of the different things that can happen here.

Reply:
The if statements reads: If the TV was off and the screen saver has turned off and the currently selected input on the tv is not our input, the select the input that we are connected to. We previously turned on the screen so we don't need to do that. We need to do it this way because we cannot get the currently selected input while the TV is off - I thought I had a comment for that - it seems to have gotten lost with some rewriting. I'll add something back.

Reply:
I have updated this function, adding a bunch of comments. Let me know how it reads now. Happy to revise some more if needed.